src/GitHubCreateMergePRs/Program.cs

```diff
@@ -157,4 +156,53 @@ private static async Task RunAsync(XDocument config, bool isAutomatedRun, bool i
             }
         }
     }
+
+    private static bool ShouldRunMerge(XElement merge, bool isAutomatedRun)
+    {
+        // We always run when merges are started manually
+        if (!isAutomatedRun)
+        {
+            return true;
+        }
+
+        var frequency = merge.Attribute("frequency")?.Value.ToLower();
+
+        // We always run when a frequency isn't specified
+        if (string.IsNullOrEmpty(frequency))
+        {
+            return true;
+        }
+
+        // We always run when an unexpected frequency value is specified
+        if (frequency != "daily" && frequency != "weekly")
+        {
+            Console.WriteLine($"##vso[task.logissue type=error]Unexpected merge frequency specified ({frequency}).");
+
+            return true;
+        }
+
+        var runDateTime = DateTime.Now;
+        var currentDate = runDateTime.Date;
+
+        // Are we within the first 15 minutes of the day?
+        var fifteenMinutes = new TimeSpan(hours: 0, minutes: 15, seconds: 0);
```

Comment:
Might be good to make a note that we can do this check b/c the current pipeline cron is running every 3 hours starting at 12am (so it will run at 12am every day).